### ISSUE TITLE
[clang-tidy] misc-coroutine-hostile-raii does not detect hostile suspensions inside assignments

### ISSUE THREAD

**Issue by usx95** [llvm/llvm-project#72793]
```cpp
Gen<int> Foo() {
  {
    absl::Mutex mtx;
    int x = co_await foo_coro(1); // no warning here.
  }
  {
    absl::Mutex mtx;
    auto x = foo_coro(co_await foo_coro(1)); // no warning here.
  }
}
```
https://godbolt.org/z/4hscdaMTz
